gluten-ut/spark33/src/test/scala/org/apache/spark/sql/execution/GlutenReplaceHashWithSortAggSuite.scala

```diff
@@ -72,6 +72,7 @@ class GlutenReplaceHashWithSortAggSuite
                |   SORT BY key
                |)
                |GROUP BY key
+               |ORDER BY key
```

Comment:
Why is this change needed? Could you add some comments?

Reply:
Because repartition modification causing the result sort order not same and casuing the result not same.

Reply:
Thanks. Please document in the code as this modification makes our test sql different from the original one https://github.com/apache/spark/blob/922f5f686dc72433a9028bbe471a35a5b84f2855/sql/core/src/test/scala/org/apache/spark/sql/execution/ReplaceHashWithSortAggSuite.scala#L61-L72